examples/s3d/test_s3d.py

```diff
@@ -0,0 +1,88 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import importlib
+
+import pytest
+import torch
+from examples.s3d.s3d import BasicConv3d, S3D, S3DEncoder, SepConv3d
+from test.test_utils import assert_expected, set_rng_seed
+
+
+class TestS3D:
+    @pytest.fixture
+    def start(self):
+        set_rng_seed(1234)
+
+        def make_input_video_from_shape(shape):
+            return torch.randint(10, shape).float()
+
+        def make_input_video(num_channels=3, c_dim=1):
+            input_shape = [2, 4, 1, 5, 3]
+            input_shape[c_dim] = num_channels
+            return make_input_video_from_shape(input_shape)
+
+        return make_input_video, make_input_video_from_shape
+
+    def test_s3d_encoder(self, start):
+        make_input_video, _ = start
+        input_video = make_input_video(c_dim=-1)
+        s3d_e = S3DEncoder(trainable=True)
+        out = s3d_e(input_video)
+        expected_sum = 809.0687
+        assert_expected(
+            actual=out.shape, expected=torch.Size([2, 1024])
+        )  # batch x embedding
+        if abs(out.sum() - expected_sum) > 5e-4:
```

Comment:
Why do we need the comparison here? Is it equivalent to `assert_expected(out.sum(), expected, atol=1e-4)`?

Reply:
it's equivalent except that `assert_expected` only takes tensors. i'll use `torch.as_tensor(expected_sum)`